contracts/ScheduledPayment.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.9;
+
+import "@gnosis.pm/zodiac/contracts/core/Module.sol";
+import "@openzeppelin/contracts-upgradeable/utils/structs/EnumerableSetUpgradeable.sol";
+
+contract ScheduledPayment is Module {
+    using EnumerableSetUpgradeable for EnumerableSetUpgradeable.Bytes32Set;
+
+    event ScheduledPaymentSetup(
+        address indexed initiator,
+        address indexed owner,
+        address indexed avatar,
+        address target,
+        address config
+    );
+    event PaymentScheduled(uint256 nonce, bytes32 spHash);
+
+    event ConfigSet(address config);
+
+    address public config;
+    uint256 public nonce;
+    EnumerableSetUpgradeable.Bytes32Set private spHash;
```

Comment:
If this is a set, maybe `spHashes` plural is better, I was a bit confused reading lower down where it was referring to the singular but calling `values()`

Reply:
updated